### ISSUE TITLE
Scrolling on each matrix rather than as a set

### ISSUE THREAD

**Issue by sej7278** [squix78/MAX7219LedMatrix#3]
I've tried the AnimateText.ino and the example in README.md and they both do the same thing - the entire text scrolls on **each** matrix, rather than scrolling **across** 4 matrices.

So I end up with Demo...Demo...Demo...Demo

rather than D...e...m...o

Any ideas?

Using NodeMCU v1.0 (Amica and Doit.am) and a Wifi-Witty, Arduino 1.6.6 and 1.6.7

**Comment by benalb** [comment]
¿Have you changed "#define NUMBER_OF_DEVICES 1"?

**Comment by ozett** [comment]
with me its the same...
`
 the entire text scrolls on each matrix, rather than scrolling across 4 matrices.
`
how to change? code improvement?

![matrix](https://cloud.githubusercontent.com/assets/5937811/17785499/54000e10-6581-11e6-89b1-ef544af8e1df.png)

**Comment by dotsam** [comment]
Turns out this is the same issue I just commented on in #7, where with these boards with 4 modules, they're aligned 90º off, so the scrolling effect doesn't work right. We need something like the "vertical" parameter from this python library https://github.com/rm-hull/max7219

**Comment by ozett** [comment]
yes, seems you are right.
i did a test with python. it worked there
https://github.com/rm-hull/max7219/issues/28

but it would be wonderfull, it it would work with this lib here and the ESP8266.. :smile:

**Comment by marcelstoer** [comment]
Yep, as explained in http://frightanic.com/iot/max7219-library-nodemcu-making/ (my site) you want that _rotate_ option for the 4-in-1 modules and also an _invert_ (upside down) option if you daisy-chain manually.

**Comment by adityarao** [comment]
Team, if fixed this for 1x4 module mentioned above @ozett and @marcelstoer The modified changes are here: https://github.com/adityarao/MAX7219LedMatrix/ 

The only change required was to rotate the matrix as below in the LedMatrix::commit()  as below: 
```
void LedMatrix::commit() {
    byte index = B0000001;
    byte col; 

    if(deviceOrientation == HORIZONTAL) {
        for (col = 0; col < myNumberOfDevices * 8; col++) {
            sendByte(col / 8, col % 8 + 1, cols[col]); 
        }
    } else if(deviceOrientation == VERTICAL){ // orient the device vertically
        for (col = 0; col < myNumberOfDevices * 8; col++) {
            xcols[col] = 0; 
        }
        // little inefficient, can be enhanced, rotate the matrix !
        for (col = 0; col < myNumberOfDevices * 8; col++) {

            for(byte bits = 0; bits < 8; bits++) 
              xcols[col] |= ((cols[bits + 8*(col/8)] & (index << (col%8))) ?
                                     (B10000000 >> bits) : 0);  
            sendByte(col / 8, col % 8 + 1, xcols[col]);
        }
    }
}
```

**Comment by adityarao** [comment]
![img_20170918_202009](https://user-images.githubusercontent.com/7914293/30548387-a5fef4c2-9caf-11e7-887d-89a862cc5034.jpg)

**Comment by witnessmenow** [comment]
@adityarao nice solution!

I'll update my PR with this solution as it uses less memory
